pkg/utils/nodeclass/nodeclass.go

```diff
@@ -36,66 +37,79 @@ func New(nodeTemplate *v1alpha1.AWSNodeTemplate) *v1beta1.NodeClass {
-			AMISelectorTerms:           NewAMISelectorTerms(nodeTemplate.Spec.AMISelector),
-			AMIFamily:                  nodeTemplate.Spec.AMIFamily,
-			UserData:                   nodeTemplate.Spec.UserData,
-			Tags:                       nodeTemplate.Spec.Tags,
-			BlockDeviceMappings: lo.Map(nodeTemplate.Spec.BlockDeviceMappings, func(bdm *v1alpha1.BlockDeviceMapping, _ int) *v1beta1.BlockDeviceMapping {
-				return NewBlockDeviceMapping(bdm)
-			}),
-			DetailedMonitoring: nodeTemplate.Spec.DetailedMonitoring,
-			MetadataOptions:    NewMetadataOptions(nodeTemplate.Spec.MetadataOptions),
-			Context:            nodeTemplate.Spec.Context,
-			LaunchTemplateName: nodeTemplate.Spec.LaunchTemplateName,
-			InstanceProfile:    nodeTemplate.Spec.InstanceProfile,
+			SubnetSelectorTerms:           NewSubnetSelectorTerms(nodeTemplate.Spec.SubnetSelector),
+			OriginalSubnetSelector:        nodeTemplate.Spec.SubnetSelector,
+			SecurityGroupSelectorTerms:    NewSecurityGroupSelectorTerms(nodeTemplate.Spec.SecurityGroupSelector),
+			OriginalSecurityGroupSelector: nodeTemplate.Spec.SecurityGroupSelector,
+			AMISelectorTerms:              NewAMISelectorTerms(nodeTemplate.Spec.AMISelector),
+			OriginalAMISelector:           nodeTemplate.Spec.AMISelector,
+			AMIFamily:                     nodeTemplate.Spec.AMIFamily,
+			UserData:                      nodeTemplate.Spec.UserData,
+			Tags:                          nodeTemplate.Spec.Tags,
+			BlockDeviceMappings:           NewBlockDeviceMappings(nodeTemplate.Spec.BlockDeviceMappings),
+			DetailedMonitoring:            nodeTemplate.Spec.DetailedMonitoring,
+			MetadataOptions:               NewMetadataOptions(nodeTemplate.Spec.MetadataOptions),
+			Context:                       nodeTemplate.Spec.Context,
+			LaunchTemplateName:            nodeTemplate.Spec.LaunchTemplateName,
+			InstanceProfile:               nodeTemplate.Spec.InstanceProfile,
 		},
 		Status: v1beta1.NodeClassStatus{
-			Subnets: lo.Map(nodeTemplate.Status.Subnets, func(s v1alpha1.Subnet, _ int) v1beta1.Subnet {
-				return v1beta1.Subnet{
-					ID:   s.ID,
-					Zone: s.Zone,
-				}
-			}),
-			SecurityGroups: lo.Map(nodeTemplate.Status.SecurityGroups, func(s v1alpha1.SecurityGroup, _ int) v1beta1.SecurityGroup {
-				return v1beta1.SecurityGroup{
-					ID:   s.ID,
-					Name: s.Name,
-				}
-			}),
-			AMIs: lo.Map(nodeTemplate.Status.AMIs, func(a v1alpha1.AMI, _ int) v1beta1.AMI {
-				return v1beta1.AMI{
-					ID:           a.ID,
-					Name:         a.Name,
-					Requirements: a.Requirements,
-				}
-			}),
+			Subnets:        NewSubnets(nodeTemplate.Status.Subnets),
+			SecurityGroups: NewSecurityGroups(nodeTemplate.Status.SecurityGroups),
+			AMIs:           NewAMIs(nodeTemplate.Status.AMIs),
 		},
 		IsNodeTemplate: true,
 	}
 }
 
-func NewSubnetSelectorTerms(subnetSelector map[string]string) []v1beta1.SubnetSelectorTerm {
+func NewSubnetSelectorTerms(subnetSelector map[string]string) (terms []v1beta1.SubnetSelectorTerm) {
```

Comment:
nit: named return values

Reply:
I generally take the approach of naming the return parameters when the return value is a list and we are going to have to declare it in the body of the function anyways with a `var` statement so I personally think that this is ok here.